app/src/ui/branches/branches-container.tsx

```diff
@@ -134,9 +142,11 @@ export class BranchesContainer extends React.Component<
             allBranches={this.props.allBranches}
             recentBranches={this.props.recentBranches}
             onItemClick={this.onItemClick}
-            filterText={this.state.filterText}
-            onFilterKeyDown={this.onFilterKeyDown}
-            onFilterTextChanged={this.onFilterTextChanged}
+            filterText={this.state.branchFilterText}
+            onFilterKeyDown={this.closeFoldoutOnEsc(
+              () => this.state.branchFilterText.length === 0
```

Comment:
Having anonymous functions inline like this means that the component will always re-render since every time this function is called a new instance of this function will be instantiated. We have a [tslint rule](https://github.com/desktop/desktop/blob/a991d2a0469120b843e06950add087fd6a2be2d2/tslint-rules/reactNoUnboundDispatcherPropsRule.ts) to prevent this but it only appears to work from within `render()`

This should be `onFilterKeyDown={this.closeFoldoutOnEsc}` and `closeFoldoutOnEsc` should be a bound method (i.e. `=>`) so that it's able to access this.state